api/src/main/java/io/minio/MinioClient.java

```diff
@@ -2413,6 +2408,10 @@ public boolean hasNext() {
               populate();
```

Comment:
Can these lines be changed to 

            if (this.error != null) {
              return true;
            }
            if (this.errorIterator == null || !this.errorIterator.hasNext()) {
              populate();
            }

Reply:
Lines 2407 thru 2421 can be written more concisely as 

```
if (this.error != null) {
     return true;
}
if (this.errorIterator.hasNext()) {
     return true;
}
populate()
```

Reply:
No. This will make loose of `error`  not addressed in every `populate()` especially first run i.e. `populate()` sets error.